### PR TITLE
New ess plots

### PR DESCRIPTION
**Pull request by OriolAbril**
Add the plots for new ess (#621).

Example plots on the centered eight dataset. The default appearance is quite simple:

image

image

image

But it can be customized to look like the plots in the reference paper for example:

image

image

I have set the same default appearance for both quantile and local, differing from the paper, because I think that using a hist like plot can be misleading. Now the only difference is the ylabel, which can be confusing to someone not familiar with this kind of plots. Would you rather have different appearance between quantile and local ess plots?

### CODE REVIEW

File: arviz/plots/essplot.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        Show relative ess in plot.
+    coords : dict, optional
+        Coordinates of var_names to be plotted. Passed to `Dataset.sel`
+    figsize : tuple, optional
+        Figure size. If None it will be defined automatically.
+    textsize: float, optional
+        Text size scaling factor for labels, titles and lines. If None it will be autoscaled based
+        on figsize.
+    n_points : int
+        Number of quantiles for which to plot their quantile/local ess or number of subsets
```
Comment (aloctavodia) [inline arviz/plots/essplot.py]: Number of points for...

File: arviz/plots/essplot.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    ----------
+    data : obj
+        Any object that can be converted to an az.InferenceData object
+        Refer to documentation of az.convert_to_dataset for details
+    var_names : list of variable names, optional
+        Variables to be plotted, two variables are required.
+    kind : str, optional
+        Options: ``local``, ``quantile`` or ``change``, specify the kind of plot.
+    relative : bool
+        Show relative ess in plot.
```
Comment (aloctavodia) [inline arviz/plots/essplot.py]: Show relative ess in plot `ress = ess / N`.

File: arviz/plots/essplot.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+):
+    """Plot quantile, local or change of effective sample sizes (ESS).
+
+    Parameters
+    ----------
+    data : obj
+        Any object that can be converted to an az.InferenceData object
+        Refer to documentation of az.convert_to_dataset for details
+    var_names : list of variable names, optional
+        Variables to be plotted, two variables are required.
```
Comment (aloctavodia) [inline arviz/plots/essplot.py]: Variables to be plotted.

Comment (OriolAbril) [reply arviz/plots/essplot.py]: True, I copy-pasted too much

**Comment by OriolAbril** [comment]
I think I have not forgotten any feature. If you have time to review and it turns out I did miss something, please say so. I will start writing tests so it can be merged soon.

File: arviz/plots/essplot.py [hunk 4]
```diff
@@ -149,6 +149,8 @@ def plot_ess(
 
     data = convert_to_dataset(idata, group="posterior")
     var_names = _var_names(var_names, data)
+    n_draws = len(data.draw)
```
[the remark this answers is not in the document]
Comment (OriolAbril) [reply arviz/plots/essplot.py]: As the last value of the dimension? I think I am not following

**Comment by OriolAbril** [comment]
> One question (is that evolution calculated as `ess(ary[:i])`?

Yes, the first `i` draws (but all chains) are selected to calculate the ess evolution in ess "change"

File: arviz/plots/essplot.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahartikainen) [reply arviz/plots/essplot.py]: data.draw is coordinate, right? The draw dim should contain value n?

**Comment by ahartikainen** [comment]
Is change desciptive name? Could it be `evolution` or something similar?

File: arviz/plots/essplot.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (OriolAbril) [reply arviz/plots/essplot.py]: Ohh! okey, yes, indeed. Thanks! :)

File: arviz/plots/essplot.py [hunk 5]
```diff
@@ -164,7 +164,7 @@ def plot_ess(
             dim="ess_dim",
         )
     elif kind == "local":
-        probs = np.arange(0, 1, 1 / n_points)
+        probs = np.linspace(0, 1 - 1 / n_points, n_points)
```
Comment (ahartikainen) [inline arviz/plots/essplot.py]: Is this same as 

    np.linspace(0,1,N, endpoint=False)

(could be different)

**Comment by aloctavodia** [comment]
LGTM!

**Comment by ahartikainen** [comment]
Similar plot for mcse quantile would be cool.

**Pull request merged**